File: week-01/inversions.py
#!/usr/bin/env python

import sys

array = []
with open('./.data', 'r') as f:
    for line in f:
        array.append(int(line))

def brute(arr):
    invs = 0
    for i in range(0, len(arr)):
        for j in range(i+1, len(arr)):
            if arr[i] > arr[j]:
                arr[i], arr[j] = arr[j], arr[i]
                invs += 1
    return invs

print brute(array)

def inversions(arr):
    pass

File: week-01/inversions.rs
use std::io::prelude::*;
use std::io::{BufReader};
use std::fs::File;

fn main() {
    let mut is = Vec::with_capacity(100_000);

    let f = File::open("./.data").unwrap();
    let reader = BufReader::new(f);
    for line in reader.lines() {
        let line = line.unwrap();
        is.push(line.parse::<usize>().unwrap());
    }

    let mut invs = 0 as usize;
    for i in 0..is.len() {
        for j in i..is.len() {
            if is[i] > is[j] {
                invs += 1;
            }
        }
    }

    println!("{}", invs);
}
